flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java

```diff
@@ -459,7 +459,8 @@ public static String[] mergeListsToArray(List<String> base, List<String> append)
             ConfigOptions.key("parallelism.default")
                     .intType()
                     .defaultValue(1)
-                    .withDescription("Default parallelism for jobs.");
+                    .withDescription(
```

Comment:
are we removing  .defaultValue(1) , as from your above description - it is never used?

Reply:
No, It used in most scenes.
Personally, I want remove the the `.defaultValue(1)` but I concern the API compatibility.
I will do it if more committers suggest to remove it.

Reply:
+1 to keep the default value as it's using in some scenes.

Reply:
I would add an existing option[1] here, it's `jobmanager.scheduler`. Although, the default value is Default, but the default value will be updated to `AdaptiveBatch ` for flink batch job. The description is: 

> Determines which scheduler implementation is used to schedule tasks. If this option is not explicitly set, batch jobs will use the 'AdaptiveBatch' scheduler as the default, while streaming jobs will default to the 'Default' scheduler.

I think it's similar with `parallelism.default`, there are some special cases that don't respect the default value.

[1] https://nightlies.apache.org/flink/flink-docs-release-1.20/docs/deployment/config/#jobmanager-scheduler

Reply:
Got it.